### PR TITLE
added lb jpa impl delete put search

### PR DESCRIPTION
**Pull request by ItsUnbidden**

### CODE REVIEW

**Review by Rommelua — APPROVED** [review]

File: src/main/java/mate/academy/jvbookstore/dto/BookSearchParametersDto.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package mate.academy.jvbookstore.dto;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@EqualsAndHashCode(callSuper = false)
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/jvbookstore/dto/BookSearchParametersDto.java:7]: Why you use this annotation?

File: src/main/java/mate/academy/jvbookstore/dto/SearchParametersDto.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package mate.academy.jvbookstore.dto;
+
+public abstract class SearchParametersDto {
+    
+}
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/jvbookstore/dto/SearchParametersDto.java:5]: Why do you need empty abstract class?

File: src/main/java/mate/academy/jvbookstore/model/Book.java [hunk 3]
```diff
@@ -30,4 +36,7 @@ public class Book {
     private String description;
     
     private String coverImage;
+
+    @Column(nullable = false)
+    private boolean isDeleted = false;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/jvbookstore/model/Book.java:41]: Why you init this field?

File: src/main/java/mate/academy/jvbookstore/repository/book/specification/AuthorSpecificationProvider.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) -> 
+                root.get(getKey()).in(Arrays.stream(params).toArray());
+    }
+
+    @Override
+    public String getKey() {
+        return "author";
+    }
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/jvbookstore/repository/book/specification/AuthorSpecificationProvider.java:21]: Use constant

File: src/main/java/mate/academy/jvbookstore/repository/book/specification/PriceSpecificationProvider.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        final Double priceFrom;
+        final Double priceTo;
+
+        try {
+            priceFrom = (params.length > 0 && params[0] != null 
+                    && params[0].length() != 0) ? Double.valueOf(params[0]) : 0.0;
+            priceTo = (params.length > 1 && params[1] != null 
+                    && params[1].length() != 0) ? Double.valueOf(params[1]) : Double.MAX_VALUE;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/jvbookstore/repository/book/specification/PriceSpecificationProvider.java:21]: Magic numbers

File: src/main/java/mate/academy/jvbookstore/dto/BookSearchParametersDto.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+public class BookSearchParametersDto extends SearchParametersDto {
+    private String[] titles;
+    
+    private String[] authors;
+
+    private String[] price;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/jvbookstore/dto/BookSearchParametersDto.java:13]: Why don`t you want to use List?

**Review by Sanderfuck — COMMENTED** [review]
You PR is merged, but why before send?

**Comment by ItsUnbidden** [comment]
This task was added just before I finished the entire project. So I added all the functionality very long ago and this is why the pr is old.

**Pull request merged**